actionpack/lib/action_controller/metal/instrumentation.rb

```diff
@@ -71,6 +71,7 @@ def redirect_to(*args)
   private
 
     # A hook invoked every time a before callback is halted.
+    # TODO: rename payload[:filter] to payload[:callback]
```

Comment:
This will be really hard to rename since we can't deprecate since it is just a normal hash. Not sure if it is worth to rename. Maybe post on both keys?